sylvia-derive/src/associated_types.rs

```diff
@@ -0,0 +1,76 @@
+use proc_macro2::{Ident, TokenStream};
+use quote::quote;
+use syn::{parse_quote, ItemTrait, TraitItem, TraitItemType, WhereClause, WherePredicate};
+
+const RESERVED_TYPES: [&str; 3] = ["Error", "QueryC", "ExecC"];
+
+#[derive(Default)]
+pub struct AssociatedTypes<'a>(Vec<&'a TraitItemType>);
+
+impl<'a> AssociatedTypes<'a> {
+    pub fn new(source: &'a ItemTrait) -> Self {
+        let associated_types: Vec<_> = source
+            .items
+            .iter()
+            .filter_map(|item| match item {
+                TraitItem::Type(ty) if !RESERVED_TYPES.contains(&ty.ident.to_string().as_str()) => {
+                    Some(ty)
+                }
+                _ => None,
+            })
+            .collect();
+
+        Self(associated_types)
+    }
+
+    pub fn as_where_predicates(&self) -> Vec<WherePredicate> {
```

Comment:
Nitpick: Consider returning an iterator

Reply:
It is used only in a single place that's why I decided to make it more "specialized"